### PR TITLE
#16567 Support Apache Kyuubi (Incubating)

### LINKED ISSUE #16567: Support Apache Kyuubi (Incubating)
**Is your feature request related to a problem? Please describe.**

Apache Kyuubi (Incubating)[1] a distributed and multi-tenant gateway to provide serverless SQL on lakehouses, and DBeaver is an amazing database/data warehouse GUI client. I want to add the native support of Kyuubi in DBeaver.

**Describe the solution you'd like**

Kyuubi provides a HiveServer2 compatible thrift API and JDBC driver, so the support is simple, I proposes to refer the Apache Spark Thrift Server to change the xml configuration and add the icons file.

**Describe alternatives you've considered**

User can connect Kyuubi by using Hive JDBC driver, but Kyuubi provides it's own JDBC driver to unlock more functionalities, e.g. Hive JDBC driver supports only the default catalog, but Kyuubi JDBC driver supports multi-catalogs.

**Additional context**

image

image

image

image

image

[1] https://kyuubi.apache.org/

### PR DESCRIPTION
**Pull request by pan3793**
Hi community,

This is Cheng from Apache Kyuubi (Incubating) community, and this PR proposes to add support in DBeaver for Apache Kyuubi (Incubating).

Kyuubi[1] a distributed and multi-tenant gateway to provide serverless SQL on lakehouses, and Kyuubi provides a HiveServer2 compatible thrift API, so the support is simple, I refer the Apache Spark Thrift Server to change the xml configuration and add the icons file.

User can connect Kyuubi by using Hive JDBC driver, and Kyuubi provides it's own JDBC driver to unlock more functionalities, e.g. Hive JDBC driver supports only the default catalog, but Kyuubi JDBC driver supports multi-catalogs.

[1] https://kyuubi.apache.org/

image

image

image

image

image

### CODE REVIEW

**Comment by yanghua** [comment]
+1 for supporting the Kyuubi connector!

**Review by yaooqinn — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml [hunk 1]
```diff
@@ -755,6 +755,24 @@
...
+                    iconBig="icons/kyuubi_hive_icon_big.png"
+                    class="org.apache.kyuubi.jdbc.KyuubiHiveDriver"
+                    sampleURL="jdbc:hive2://{host}[:{port}][/{database}]"
+                    defaultPort="10009"
+                    description="Kyuubi Thrift Server connector"
+                    webURL="https://kyuubi.apache.org/"
+                    category="Hadoop"
+                    categories="hadoop">
+                    <file type="jar" path="maven:/org.apache.kyuubi:kyuubi-hive-jdbc-shaded:RELEASE" bundle="!drivers.kyuubi"/>
+                    <file type="jar" path="drivers/kyuubi" bundle="drivers.kyuubi"/>
```
Comment (pan3793) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml:771]: This is for cloudbeaver, I can send PR once this get merged.

image

**Comment by pan3793** [comment]
Hi @E1izabeth and @tati-kru, I notice the previous commit comments violate the check rule, I created an issue #16567 and updated the commit comments, please take a look when you have time :)

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml (hunk 1 above)
Comment (pan3793) [reply plugins/org.jkiss.dbeaver.ext.generic/plugin.xml:771]: I found that the cloud beaver community have concernd about the driver dependencies and size.
https://github.com/dbeaver/cloudbeaver/issues/178#issuecomment-772631295

I'd like to clarify that Kyuubi Hive JDBC driver is based on the Hive JDBC driver and we are keep improving it. Currently, we shaded and relocated all transitive classes except to Hadoop(for Kerberos) and SLF4J. And we have a plan to rewrite this Driver totally to make it light and zero-deps in the future.

**Comment by pan3793** [comment]
Sorry, I have no idea why "Commit Message Validation" fails consistently, I think the commit message should meet the requirements now. Do you think it's a blocking issue?

**Comment by E1izabeth** [comment]
@pan3793 don't worry about it, we might have problem with Commit Validation check, because it still checks your first commit message

**Comment by AngersZhuuuu** [comment]
+1 for this

**Comment by Matvey16** [comment]
Verified

**Comment by pan3793** [comment]
Thanks everyone for helping review and verify!

**Pull request merged**